pyaedt/twinbuilder.py

```diff
@@ -309,5 +312,258 @@ def set_sim_setup_parameter(self, var_str, expression, analysis_name="TR"):
+        z_ref=50,
+        state_space_dynamic_link_type=None,
+        component_name=None,
+        save_project=True,
+    ):
+        """Add a Q2D or Q3D dynamic component to a Twin Builder design.
+
+        Parameters
+        ----------
+        source_project : str
+            Source project name or project path.
+        source_design_name : str
+            Source Design name in specified project.
+        setup_name : str
+            Setup name.
+        sweep_name : str
+            Sweep name
+        coupling_matrix_name : str
+            Coupling matrix name.
+        model_depth : str, optional
+            2D model depth specified as value with unit.
+            To be provided if design is Q2D.
+            The default value is ``1meter``
+        maximum_order : float, optional
+            The Maximum Order value specifies the highest order state space
+            system that you can choose while fitting to represent the system behavior.
+            A lower order may lead to less accuracy but faster simulation.
+            The default value is ``10000``.
+        error_tolerance : float, optional
+            Error Tolerance sets the error tolerance for S-Matrix fitting.
+            The default value is ``0.005``.
+        z_ref : float, optional
+            Sets the value of the Z (ref) in ohms.
+            The default value is ``50``.
+        state_space_dynamic_link_type : str, optional
+            Q3D state space dynamic link type.
+            Possible options are:
+                - ``S`` for S parameters link type.
+                - ``RLGC`` for RLGC Parameters link type.
+                - ``EQ`` for Equivalent Circuit.
+            The default value is ``RLGC``.
+        component_name : str, optional
+            Component name.
+            If not provided a generic name with prefix "SimpQ3DData" will be given.
+        save_project : bool, optional
+            Save project after use.
+            The default value is ``True``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+        >>> oComponentManager.AddDynamicNPortData
+
+        Examples
+        --------
+        Create an instance of Twin Builder.
+
+        >>> from pyaedt import TwinBuilder
+        >>> tb = TwinBuilder()
+
+        Add a Q2D dynamic link component.
+
+        >>> tb.add_q3d_dynamic_component("Q2D_ArmouredCableExample",
+        ...                             "2D_Extractor_Cable",
+        ...                             "MySetupAuto",
+        ...                             "sweep1",
+        ...                             "Original",
+        ...                             "100mm")
+        >>> tb.release_desktop()
+        """
+        dkp = self.desktop_class
+        is_loaded = False
+        app = None
+        if os.path.isfile(source_project):
+            if os.path.exists(source_project):
```

Comment:
If you valid os.path.isfile i assume that the second if statement isn't needed ?